### DISCUSSION TITLE
How to add transition for `bg-gradient`

### DISCUSSION THREAD

**Discussion by amrap030** [tailwindlabs/tailwindcss#6154] [Help] [answered]
Hello together,

I would like to know, how to add a transition from a white background to a gradient background. My current code looks like the following:

```html
<li class="transition duration-500 ease-in-out rounded-lg  hover:shadow-lg bg-gradient-to-br hover:from-cyan-400 hover:to-light-blue-500 hover:text-gray-50">
...
</li>
```

Thanks in advance for your help!

**Comment by adamwathan** [comment, marked as answer]
Hey! Unfortunately CSS doesn't make it possible to transition background images (which is how gradients are implemented), so the trick is to transition the opacity on a separate layer:

https://play.tailwindcss.com/rx9UOa1bD1

**Reply by Chandler-Zhu** [reply]
In the demo there is a `relative` class for the text, when I remove this class, the text becomes invisible in hover state. Could you explain this behavior to me ?

**Comment by PositiveUser** [comment]
You can achieve this using before pseudo element

Check this [Tailwind Play](https://play.tailwindcss.com/U44mKElw8v)

Thank you very much
